### PR TITLE
refactor(artifact): remove oci artifact support

### LINKED ISSUE #1425: [Feat]: Add support for artifactType
### Is your feature request related to a problem? Please describe.

The `artifactType` field was added in the latest RC's from OCI. Changes are needed in Zot to support this field.

### Describe the solution you'd like

The field is added in [image-spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md).

The referrers response in [distribution-spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#push) was updated to prefer the `artifactType` when it's available:

> The value of the `artifactType` MUST be set to the `artifactType` value in the image manifest, if present. If the `artifactType` is empty or missing in the image manifest, the value of `artifactType` MUST be set to the config descriptor `mediaType` value. 

### Describe alternatives you've considered

_No response_

### Additional context

```
$ regctl manifest get --format '{{jsonPretty .}}'  localhost:5000/regclient/regctl@sha256:9180968bc41e484
d35ffab7a457215ffde92b6df7846a93d0d42df258369f415
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "artifactType": "application/spdx+json",
  "config": {
    "mediaType": "application/vnd.oci.scratch.v1+json",
    "size": 2,
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a"
  },
  "layers": [
    {
      "mediaType": "application/spdx+json",
      "size": 8813,
      "digest": "sha256:6d3d8f08665edbc2e689f733a97a1d55e910df54e29807edff84f37b0acccc5d"
    }
  ],
  "annotations": {
    "org.opencontainers.artifact.created": "2023-05-04T00:37:17Z",
    "org.opencontainers.artifact.description": "SPDX JSON SBOM"
  },
  "subject": {
    "mediaType": "application/vnd.oci.image.manifest.v1+json",
    "size": 1024,
    "digest": "sha256:e1d228201ef3eb78a5d2be08f2bd7bb61bed00d6dc551e3a2eaf382fa417d6d4"
  }
}

$ # CNCF distribution on port 5000

$ regctl artifact list localhost:5000/regclient/regctl:edge --platform local --format '{{jsonPretty .Manifest}}'
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.index.v1+json",
  "manifests": [
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "size": 698,
      "digest": "sha256:be45432ab8a3972416b73e310229f46434f4700026d33d10a710c193a48310ee",
      "annotations": {
        "vnd.docker.reference.type": "attestation-manifest"
      },
      "artifactType": "application/vnd.oci.image.config.v1+json"
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "size": 736,
      "digest": "sha256:67c41fa76a0b9543477b4a19bd5b8d283eb13b92c01ce7942c38929ba9f97999",
      "annotations": {
        "org.opencontainers.artifact.created": "2023-05-04T00:37:17Z",
        "org.opencontainers.artifact.description": "CycloneDX JSON SBOM"
      },
      "artifactType": "application/vnd.cyclonedx+json"
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "size": 712,
      "digest": "sha256:9180968bc41e484d35ffab7a457215ffde92b6df7846a93d0d42df258369f415",
      "annotations": {
        "org.opencontainers.artifact.created": "2023-05-04T00:37:17Z",
        "org.opencontainers.artifact.description": "SPDX JSON SBOM"
      },
      "artifactType": "application/spdx+json"
    }
  ]
}

$ # zot on port 5002

$ regctl artifact list localhost:5002/regclient/regctl:edge --platform local --format '{{jsonPretty .Mani
fest}}'
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.index.v1+json",
  "manifests": [
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:9180968bc41e484d35ffab7a457215ffde92b6df7846a93d0d42df258369f415",
      "size": 712,
      "annotations": {
        "org.opencontainers.artifact.created": "2023-05-04T00:37:17Z",
        "org.opencontainers.artifact.description": "SPDX JSON SBOM"
      },
      "artifactType": "application/vnd.oci.scratch.v1+json"
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:67c41fa76a0b9543477b4a19bd5b8d283eb13b92c01ce7942c38929ba9f97999",
      "size": 736,
      "annotations": {
        "org.opencontainers.artifact.created": "2023-05-04T00:37:17Z",
        "org.opencontainers.artifact.description": "CycloneDX JSON SBOM"
      },
      "artifactType": "application/vnd.oci.scratch.v1+json"
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:be45432ab8a3972416b73e310229f46434f4700026d33d10a710c193a48310ee",
      "size": 698,
      "annotations": {
        "vnd.docker.reference.type": "attestation-manifest"
      },
      "artifactType": "application/vnd.oci.image.config.v1+json"
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:
Closes #1425

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by peusebiu** [comment]
we also have specific logic for this mediatype in sync: 
https://github.com/project-zot/zot/blob/main/pkg/extensions/sync/utils.go#L682

**Comment by rchincha** [comment]
https://github.com/regclient/regclient
^ build from top of main for the new image spec functionality

**Comment by rchincha** [comment]
If filtering is requested and applied, the response MUST include a header OCI-Filters-Applied: artifactType denoting that an artifactType filter was applied. If multiple filters are applied, the header MUST contain a comma separated list of applied filters.

**Comment by rchincha** [comment]
https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers

**Comment by rchincha** [comment]
https://github.com/project-zot/zot/issues/1406

File: pkg/cli/cve_cmd_test.go [hunk 1]
```diff
@@ -713,7 +713,7 @@ func TestServerCVEResponse(t *testing.T) {
 		str := space.ReplaceAllString(buff.String(), " ")
 		str = strings.TrimSpace(str)
 		So(err, ShouldBeNil)
-		So(str, ShouldEqual, "IMAGE NAME TAG OS/ARCH DIGEST SIGNED SIZE zot-cve-test 0.0.1 82836dd7 false 548B")
+		So(str, ShouldEqual, "IMAGE NAME TAG OS/ARCH DIGEST SIGNED SIZE zot-cve-test 0.0.1 40d1f749 false 605B")
```
Comment (andaaron) [inline pkg/cli/cve_cmd_test.go:716]: Why do the sizes and digests change?

File: pkg/meta/repodb/storage_parsing.go [hunk 2]
```diff
@@ -335,43 +320,22 @@ func GetReferredSubject(descriptorBlob []byte, referrerDigest, mediaType string,
...
-
-		err := json.Unmarshal(descriptorBlob, &manifestContent)
-		if err != nil {
-			return "", referrerInfo, false,
-				fmt.Errorf("repodb: can't unmarhsal artifact manifest for digest %s: %w", referrerDigest, err)
-		}
+	err := json.Unmarshal(descriptorBlob, &manifestContent)
+	if err != nil {
+		return "", referrerInfo, false,
+			fmt.Errorf("repodb: can't unmarhsal manifest for digest %s: %w", referrerDigest, err)
```
Comment (andaaron) [inline pkg/meta/repodb/storage_parsing.go:328]:
```suggestion
			fmt.Errorf("repodb: can't unmarshal manifest for digest %s: %w", referrerDigest, err)
```

**Review by andaaron — APPROVED** [review]

**Comment by laurentiuNiculae** [comment]
Results for some tests I did using latest regctl version: [link](https://gist.github.com/laurentiuNiculae/c570779412d96a252db41f1197661a8f)

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**